bin/roave-infection-static-analysis-plugin

```diff
@@ -56,10 +57,61 @@ use function var_export;
         define('PHP_PARSER_VERSION', Versions::getVersion('nikic/php-parser'));
     }
 
-    $makeAnalyzer = static function () use ($projectPath): ProjectAnalyzer {
+    $extractArgument =
+        /**
+         * @param list<string> $arguments
+         *
+         * @return array{0: list<string>, 1: string|null}
+         */
+        static function(array $arguments, string $arg): array {
+            $lookup = '--'.$arg;
+
+            $result = null;
+            foreach ($arguments as $index => $argument) {
```

Comment:
This code is really ugly as heck - is there a way to simplify it? 🤔

Reply:
any better ideas? 😅  couldn't really come up with anything better personally 🤷‍♂️

Reply:
I'd say:

 1. Isolate+test
 2. Refactoring attempts after that